cmd/config-v17.go

```diff
@@ -91,67 +162,25 @@ func newConfig() error {
 	// Initialize server config.
 	srvCfg := newServerConfigV17()
 
-	// If env is set for a fresh start, save them to config file.
+	// If env is set override the credentials from config file.
 	if globalIsEnvCreds {
 		srvCfg.SetCredential(globalActiveCred)
 	}
-
 	if globalIsEnvBrowser {
 		srvCfg.SetBrowser(globalIsBrowserEnabled)
 	}
 
-	// Create config path.
-	if err := createConfigDir(); err != nil {
-		return err
-	}
```

Comment:
I see that `createConfigDir()` is getting called in gateway mode, but unable to see it getting called for server mode